runtime/lua/vim/lsp/buf.lua

```diff
@@ -1311,4 +1311,19 @@ function M.execute_command(command_params)
   lsp.buf_request(0, ms.workspace_executeCommand, command_params)
 end
 
+--- @class vim.lsp.WorkspaceDiagnosticsOpts
+--- @inlinedoc
+---
+--- Only request diagnostics from the indicated client. If nil, the request is sent to all clients.
+--- @field client_id? integer
+
+--- Request workspace-wide diagnostics.
+--- @param opts? vim.lsp.WorkspaceDiagnosticsOpts
+--- @see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_dagnostics
+function M.workspace_diagnostics(opts)
```

Comment:
nit: place it near the other workspace functions? https://github.com/neovim/neovim/blob/96a0e5f265d11e7913ce71a3a1981b267bda79e5/runtime/lua/vim/lsp/buf.lua#L1009

Reply:
Sure.